x-pack/metricbeat/module/azure/monitor/client_helper.go

```diff
@@ -23,9 +27,37 @@ func mapMetrics(client *azure.Client, resources []*armresources.GenericResourceE
 	for _, resource := range resources {
 		for _, metric := range resourceConfig.Metrics {
 			// get all metrics supported by the namespace provided
-			metricDefinitions, err := client.AzureMonitorService.GetMetricDefinitions(*resource.ID, metric.Namespace)
-			if err != nil {
-				return nil, fmt.Errorf("no metric definitions were found for resource %s and namespace %s %w", *resource.ID, metric.Namespace, err)
+			var metricDefinitions armmonitor.MetricDefinitionCollection
+			var err error
+
+			for {
+				metricDefinitions, err = client.AzureMonitorService.GetMetricDefinitions(*resource.ID, metric.Namespace)
+				if err != nil {
+					errorMsg := "no metric definitions were found for resource " + *resource.ID + " and namespace " + metric.Namespace
+
+					var respError *azcore.ResponseError
+					ok := errors.As(err, &respError)
+					if !ok {
+						return nil, fmt.Errorf("%s, %w", errorMsg, err)
```

Comment:
I think this should be a different error message and not errorMsg, sth like the 'The returned error does not match *azcore.ResponseError target'

Reply:
I believe it is impossible for this error to happen

Reply:
Fixed, added new message in latest commit